src/core/dbus-service.c

```diff
@@ -306,6 +308,9 @@ static int bus_service_set_transient_property(
         if (streq(name, "WatchdogUSec"))
                 return bus_set_transient_usec(u, name, &s->watchdog_usec, message, flags, error);
 
+        if (streq(name, "WatchdogSignal"))
+                return bus_set_transient_int(u, name, &s->watchdog_signal, message, flags, error);
```

Comment:
OK. Now, the value `watchdog_signal` is not checked in `service_verify()`. So, please also check the input here. I guess `BUS_DEFINE_SET_TRANSIENT_IS_VALID()` macro can be used to define a function suitable here.